addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/model/HttpChannelConfig.java

```diff
@@ -0,0 +1,263 @@
+            return username;
+        }
+
+        public Optional<String> getPassword() {
+            return password;
+        }
+
+        public Duration getConnectTimeout() {
+            return connectTimeout;
+        }
+
+        public Duration getRequestTimeout() {
+            return requestTimeout;
+        }
+
+        public Duration getRefreshInterval() {
+            return refreshInterval;
+        }
+
+        public Optional<Transform> getResponseTransform() {
+            return responseTransform;
+        }
+    }
+
+    /**
+     * A class describing configuration for the HTTP request to make when sending a {@link Command}.
+     */
+    public static class CommandRequest {
+        private final Method method;
+        private final URL url;
+        private final Optional<String> username;
+        private final Optional<String> password;
+        private final Duration connectTimeout;
+        private final Duration requestTimeout;
+        private final String contentType;
+        private final Optional<Transform> requestTransform;
+        private final Optional<Transform> responseTransform;
+
+        CommandRequest(final Method method,
+                       final URL url,
+                       final Optional<String> username,
+                       final Optional<String> password,
+                       final Duration connectTimeout,
+                       final Duration requestTimeout,
+                       final String contentType,
+                       final Optional<Transform> requestTransform,
+                       final Optional<Transform> responseTransform)
+        {
+            this.method = method;
+            this.url = url;
+            this.username = username;
+            this.password = password;
+            this.connectTimeout = connectTimeout;
+            this.requestTimeout = requestTimeout;
+            this.contentType = contentType;
+            this.requestTransform = requestTransform;
+            this.responseTransform = responseTransform;
+        }
+
+        public Method getMethod() {
+            return method;
+        }
+
+        public URL getUrl() {
+            return url;
+        }
+
+        public Optional<String> getUsername() {
+            return username;
+        }
+
+        public Optional<String> getPassword() {
+            return password;
+        }
+
+        public Duration getConnectTimeout() {
+            return connectTimeout;
+        }
+
+        public Duration getRequestTimeout() {
+            return requestTimeout;
+        }
+
+        public String getContentType() {
+            return contentType;
+        }
+
+        public Optional<Transform> getRequestTransform() {
+            return requestTransform;
+        }
+
+        public Optional<Transform> getResponseTransform() {
+            return responseTransform;
+        }
+    }
+
+    @SuppressWarnings("unused")
+    private @Nullable String stateUrl;
```

Comment:
If you ever need to suppress the unused warning, something is wrong in your code.
And the static analyser is right here, because you assign to the "url" field and you are using the "stateurl" and "commandurl" field. Please check all unused fields in this class.

Reply:
Now that I re-read the class, I have realized the indentation level and that `url` and `stateurl`/`commandurl` are not in one class. I think it would help reviewing and understanding the code if you could move the two inner classes into their own files

Reply:
I'll do the refactor (agree that the inner classes make it confusing), but the reason the warnings show up is because the static analyzer / compiler can't tell that `Configuration.as(Class<?> cls)` uses reflection to assign values to those fields.  In this case the warning is entirely spurious.  (It's actually not an "unused variable" warning; it's a "private field is never assigned" warning, which weirdly can be suppressed by suppressing the "unused" warning.)